model/src/main/java/io/spine/validation/RequiredPolicy.java

```diff
@@ -65,15 +67,24 @@
                                           this);
             return EitherOf2.withA(requiredRule(declaration, field.get()));
         }
-        return EitherOf2.withB(nothing());
+        return withNothing();
     }
 
-    private static SimpleRuleAdded requiredRule(Field declaration, RequiredField field) {
-        SimpleRule rule = RequiredRule.forField(declaration, field.getErrorMessage());
-        return SimpleRuleAdded
-                .newBuilder()
-                .setType(declaration.getDeclaringType())
-                .setRule(rule)
-                .vBuild();
+    private static RuleAdded requiredRule(Field declaration, RequiredField field) {
+        Rule rule = RequiredRule.forField(declaration, field.getErrorMessage())
+                .orElseThrow(() -> doesNotSupportRequired(declaration));
+        return Rules.toEvent(rule, declaration.getDeclaringType());
+    }
+
+    private static IllegalStateException doesNotSupportRequired(Field field) {
+        String fieldName = field.getName()
+                                .getValue();
+        String typeUrl = typeUrl(field.getDeclaringType());
+        PrimitiveType type = field.getType()
+                                  .getPrimitive();
+        return new IllegalStateException(format(
```

Comment:
We have `Exceptions.newIllegalStateException()` which does exactly this. Why not use it?